snuba/utils/streams/processing/strategies/streaming/transform.py

```diff
@@ -118,7 +118,7 @@ def __len__(self) -> int:
     def get_content_size(self) -> int:
         return self.__offset
 
-    def __getitem__(self, index: int) -> Message[TPayload]:
+    def __getitem__(self, index: int) -> Any:
```

Comment:
Wouldn't this type of change cause problems for any callers of this method that rely on a tighter type bounds on the return type? I'd expect issues with `__iter__` below as it now claims to return a specified type that is no longer specified by `__getitem__` through the index access on line 161. It doesn't seem so much as this is fixing an error as moving it to a different location, and I'd expect that to reduce net typing coverage.